src/client.rs

```diff
@@ -74,6 +76,164 @@ pub extern "C" fn rustls_client_config_builder_build(
+            end_entity_cert_der: end_entity.into(),
+            intermediate_certs_der: &intermediates,
+            dns_name: dns_name.into(),
+            ocsp_response: ocsp_response.into(),
+        };
+        let result: rustls_result = unsafe { cb(self.userdata, &params) };
+        match result {
+            rustls_result::Ok => Ok(ServerCertVerified::assertion()),
+            r => match result_to_tlserror(&r) {
+                error::Either::TLSError(te) => Err(te),
+                error::Either::String(se) => Err(TLSError::General(se)),
+            },
+        }
+    }
+}
+
+/// Set a custom server certificate verifier.
+///
+/// The userdata pointer must stay valid until (a) all sessions created with this
+/// config have been freed, and (b) the config itself has been freed.
+/// The callback must not capture any of the pointers in its
+/// rustls_verify_server_cert_params.
+///
+/// The callback must be safe to call on any thread at any time, including
+/// multiple concurrent calls. So, for instance, if the callback mutates
+/// userdata (or other shared state), it must use synchronization primitives
+/// to make such mutation safe.
+///
+/// The callback receives certificate chain information as raw bytes.
+/// Currently this library offers no functions for C code to parse the
+/// certificates, so it's only possible to implement verifiers that either
+/// (a) always succeed (or fail), or (b) compare the certificates against
+/// static bytes. We plan to export parsing code in the future to make it
+/// possible to implement other strategies.
+///
+/// If you intend to write a verifier that accepts all certificates, be aware
+/// that special measures are required for IP addresses. Rustls currently
+/// (0.19.0) doesn't support building a ClientSession with an IP address
+/// (because it's not a valid DNSNameRef). One workaround is to detect IP
+/// addresses and rewrite them to `example.invalid`, and _also_ to disable
+/// SNI via rustls_client_config_builder_set_enable_sni (IP addresses don't
+/// need SNI).
+///
+/// If the custom verifier accepts the certificate, it should return
+/// RUSTLS_RESULT_OK. Otherwise, it may return any other rustls_result error.
+/// Feel free to use an appropriate error from the RUSTLS_RESULT_CERT_*
+/// section.
+///
+/// https://docs.rs/rustls/0.19.0/rustls/struct.DangerousClientConfig.html#method.set_certificate_verifier
+#[no_mangle]
+pub extern "C" fn rustls_client_config_builder_dangerous_set_certificate_verifier(
+    config: *mut rustls_client_config_builder,
+    callback: rustls_verify_server_cert_callback,
+    userdata: rustls_verify_server_cert_user_data,
+) {
+    ffi_panic_boundary_unit! {
+        let callback: VerifyCallback = match callback {
+            Some(cb) => cb,
+            None => return,
+        };
+        let config: &mut ClientConfig = try_ref_from_ptr!(config, &mut ClientConfig, ());
+        let verifier: Verifier = Verifier{callback: callback, userdata};
```

Comment:
```suggestion
        let verifier = Verifier{callback, userdata};
```
I think the `Verifier` literal is clear enough without the type annotation.